jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -2007,4 +2004,87 @@ private IllegalCharacterException(State state, HttpTokens.Token token, ByteBuffe
                 LOG.debug(String.format("Illegal character %s in state=%s for buffer %s", token, state, BufferUtil.toDetailString(buffer)));
         }
     }
+
+    private static class FieldCache
+    {
+        private int _size = 1024;
+        private Index.Mutable<HttpField> _cache;
+        private List<HttpField> _cacheableFields;
+        private boolean _caseSensitive;
+
+        public int getCapacity()
+        {
+            return _size;
+        }
+
+        public void setCapacity(int size)
+        {
+            _size = size;
+            if (_size <= 0)
+                _cache = NO_CACHE;
+            else
+                _cache = null;
+        }
+
+        public boolean isCaseSensitive()
+        {
+            return _caseSensitive;
+        }
+
+        public void setCaseSensitive(boolean caseSensitive)
+        {
+            _caseSensitive = caseSensitive;
+        }
+
+        public boolean isEnabled()
+        {
+            return _cache != NO_CACHE;
+        }
+
+        public Index<HttpField> getCache()
+        {
+            return _cache;
+        }
+
+        public HttpField getBest(ByteBuffer buffer, int i, int remaining)
+        {
+            Index.Mutable<HttpField> cache = _cache;
+            return cache == null ? null : _cache.getBest(buffer, i, remaining);
+        }
+
+        public void add(HttpField field)
+        {
+            if (_cache == null)
+            {
+                if (_cacheableFields == null)
+                    _cacheableFields = new ArrayList<>();
+                _cacheableFields.add(field);
+            }
+            else if (!_cache.put(field))
+            {
+                _cache.clear();
+                _cache.put(field);
+            }
+        }
+
+        public boolean cacheable(HttpHeader header, String valueString)
+        {
+            return isEnabled() && header != null && valueString.length() <= _size;
+        }
+
+        private void prepare()
+        {
+            if (_cache == null && _cacheableFields != null)
+            {
+                _cache = Index.buildMutableVisibleAsciiAlphabet(_caseSensitive, _size);
+                for (HttpField f : _cacheableFields)
+                {
+                    if (!_cache.put(f))
+                        break;
+                }
+                _cacheableFields.clear();
```

Comment:
This call to `clear` is useless as you're nulling out the ref on the next line.

Reply:
no harm helping out the GC.